In [1]:
import pandas as pd
import plotly.express as px

In [2]:
df = pd.read_csv("Compute_allocations_transformer.csv")
# byte_level = df[df['Patch size'] == 1]
byte_level = df[(df['FLOPs'] == 's') & (df['Patch size'] == 2)]
# byte_level['Global transformer parameters'].sort_values(ascending=False)
fig = px.histogram(byte_level['Global model parameters'])

fig.show()

In [8]:
byte_level[(byte_level['Global model parameters'] <= 1.9e7) & (byte_level['Global model parameters'] >= 1.5e7)]

,FLOPs,Patch size,Tokens,Encoder parameters,Global model parameters,Decoder parameters,Global model dimension,Encoder/Decoder dimension,Global model layers,Decoder layers
2,s,2,4.106605e+10,787087.36,15938519.04,1574174.72,576,256,4,2
56,s,2,4.001591e+10,787087.36,15741747.20,1574174.72,512,256,5,2
116,s,2,3.457268e+10,787087.36,18890096.64,1574174.72,512,256,6,2
175,s,2,3.622067e+10,787087.36,16873185.28,1574174.72,448,256,7,2
296,s,2,3.585409e+10,787087.36,15938519.04,1574174.72,384,256,9,2
353,s,2,3.289585e+10,787087.36,17709465.60,1574174.72,384,256,10,2
541,s,2,3.743909e+10,442736.64,15938519.04,1328209.92,384,192,9,3
706,s,2,3.458991e+10,442736.64,15987712.00,1328209.92,320,192,13,3
810,s,2,3.604389e+10,442736.64,15938519.04,1770946.56,384,192,9,4
852,s,2,3.305556e+10,442736.64,17709465.60,1770946.56,384,192,10,4


4e18 FLOPs
===

tokens, global model params, decoder params, global model dimension, encoder/decoder dimension, global model layers, decoder layers

Patch size = 2

5.509120e+10, 10625679,	885473,	384, 192, 6, 2 (10M run)

1.993053e+10 31483494, 3689472, 512, 320, 10, 3 (30M run)

3.622067e+10, 16873185, 1574175, 448, 256, 7, 2 (17M)

Patch size = 4

4.937337e+10, 21691515, 1836319, 576, 256, 7, 3

1.847138e+10, 60177695, 5508956, 704, 384, 13, 4

9.375604e+09, 120M, 12242125, 896, 512, 16, 5